index.bs

```diff
@@ -1000,272 +984,134 @@ that user agents must support on {{FileReader}} as DOM attributes:
-    or a {{Blob}} argument not obtained from the underlying OS file system.
-</dl>
-
-
-## Determining Encoding ## {#enctype}
-
-When reading {{Blob}} objects using the {{FileReader/readAsText()}} <a>read method</a>,
-the following <dfn id="encoding-determination">encoding determination</dfn> steps must be followed:
-
-1. Let |encoding| be null.
-2. If the {{FileReader/readAsText()/encoding}} argument is present when calling the method,
-  set |encoding| to the result of the <a>getting an encoding</a> from {{FileReader/readAsText()/encoding}}.
-3. If the <a>getting an encoding</a> steps above return failure,
-  then set |encoding| to null.
-4. If |encoding| is null,
-  and the {{FileReader/readAsText()/blob}} argument's {{Blob/type}} attribute is present,
-  and it uses a Charset Parameter [[RFC2046]],
-  set |encoding| to the result of <a>getting an encoding</a>
-  for the portion of the Charset Parameter that is a <i>label</i> of an encoding.
+   or if {{FileReader/readyState}} = {{FileReader/DONE}}
+   set {{FileReader/result}} to `null`
+   and [=terminate this algorithm=].
+1. If {{FileReader/readyState}} = {{FileReader/LOADING}}
+   set {{FileReader/readyState}} to {{FileReader/DONE}}
+   and {{FileReader/result}} to `null`.
+1. If there are any [=tasks=] from the [=context object=]
+   on the [=file reading task source=] in an affiliated [=queue a task|task queue=],
+   then remove those [=tasks=] from that task queue.
+1. [=terminate an algorithm|Terminate the algorithm=] for the [=read method=] being processed.
+1. [=Fire a progress event=] called {{abort}} at the [=context object=].
+1. If [=context object=]'s {{FileReader/readyState}} is not {{LOADING}},
+   [=fire a progress event=] called {{loadend}} at the [=context object=].
+
+## Packaging data ## {#packaging-data}
+
+<div algorithm="package data">
+A {{Blob}} has an associated <dfn for=Blob>package data</dfn> algorithm,
+given |bytes|, a |type|, a optional |mimeType| and a optional |encodingName|,
+switches on |type|, and runs the associated steps:
+
+: DataURL
+:: Return |bytes| as a DataURL [[!RFC2397]] subject to the considerations below:
+
+  * Use |mimeType| as part of the Data URL if it is available
+    in keeping with the Data URL specification [[!RFC2397]].
+  * If |mimeType| is not available return a Data URL without a media-type. [[!RFC2397]].
+    Data URLs that do not have media-types [[RFC2046]]
+    must be treated as plain text by conforming user agents. [[!RFC2397]].
+
+: Text
+:: 1. Let |encoding| be null.
+   1. If the |encodingName| is present, set |encoding| to the result of
+      [=getting an encoding=] from |encodingName|.
+   1. If the [=getting an encoding=] step above returns failure, set |encoding| to null.
+   1. If |encoding| is null, and |mimeType| is present:
+     1. Let |type| be the result of [=parse a MIME type=] given |mimeType|.
+     1. If |type| is failure, abort these substeps.
```

Comment:
Might be good to file a follow-up bug on removing this kind of anti-pattern. A preferred style would be checking that type is not failure and then nesting the further set of steps, or something equivalent.